File: src/main/scala/com/vc/easy/L283.scala
package com.vc.easy

object L283 {
  def moveZeroes(nums: Array[Int]): Unit = {
    val n = nums.length
    var i = 0
    var j = 0
    while(i < n) {
      if(nums(i) != 0) {
        nums(j) = nums(i)
        j += 1
      }
      i += 1
    }
    (j until n).foreach(i => nums(i) = 0)
  }
}

File: src/main/java/com/vc/easy/L283.java
package com.vc.easy;

class L283 {
    public void moveZeroes(int[] nums) {
        int insertIndex = 0, i = 0;
        while(i < nums.length) {
            if(nums[i] != 0) nums[insertIndex++] = nums[i];
            i++;
        }

        while(insertIndex < nums.length) {
            nums[insertIndex] = 0;
            insertIndex++;
        }
    }
}
